pyproject.toml

```diff
@@ -17,6 +17,10 @@ classifiers = [
     "Operating System :: OS Independent",
 ]
 
+dependencies = [
+    "typing-extensions>=4.4.0; python_version < '3.10'",
+]
```

Comment:
why do we need this? atm I think it's only ever used within `if TYPE_CHECKING` blocks, so we shouldn't need it

Reply:
I think I added when trying another implementation. Got rid of it! Thanks for spotting